lib/eslint.js

```diff
@@ -600,6 +617,9 @@ module.exports = (function() {
         // process initial config to make it safe to extend
         config = prepareConfig(config || {});
 
+        // check config for missing rules
+        verifyRulesExist(config);
```

Comment:
My thinking there was that it was more readable and modular to get these rules checks out of the way first than to if/else depending on whether a rule exists during the main routine.

I can definitely combine them though and see how it looks, perhaps it won't be so bad.